### DISCUSSION TITLE
ltsp-initrd-bottom permission denied

### DISCUSSION THREAD

**Discussion by Akshay-Hegde** [ltsp/ltsp#396] [General]
Strange error, I am able to login through second menu where just cli available, can someone tell cause for below error

![Untitled](https://user-images.githubusercontent.com/7464492/105036283-4aac8280-5a82-11eb-9057-b0f5a7dfeb06.jpeg)

My environment

```
root@akshay-VirtualBox:/srv/ltsp# lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.1 LTS
Release:	20.04
Codename:	focal
root@akshay-VirtualBox:/srv/ltsp# cat /etc/issue
Ubuntu 20.04.1 LTS \n \l

root@akshay-VirtualBox:/srv/ltsp# ltsp info
LTSP version: 21.01-1~ubuntu20.04.1

SERVER XSESSIONS: mate.desktop

SERVER OS: PRETTY_NAME="Ubuntu 20.04.1 LTS"

CHROOTS:
x86_64

VMs:


IMAGES:
x86_64

root@akshay-VirtualBox:/srv/ltsp# showmount -e
Export list for akshay-VirtualBox:
/srv/tftp/ltsp   *
/srv/ltsp        *
/srv/ltsp/x86_64 *


root@akshay-VirtualBox:/srv/ltsp# ll /srv/tftp/ltsp/
total 1488
drwxr-xr-x 3 root root   4096 Jan 19 17:48 ./
drwxr-xr-x 3 root root   4096 Jan 19 12:36 ../
-rw-r--r-- 1 root root 152576 Jan 19 17:48 ltsp.img
-rw-r--r-- 1 root root   3063 Jan 19 17:47 ltsp.ipxe
-rw-r--r-- 1 root root 161968 Jan  3 12:49 memtest.0
-rw-r--r-- 1 root root 930136 Jan  3 12:49 memtest.efi
-rw-r--r-- 1 root root 181952 Jan  3 12:49 snponly.efi
-rw-r--r-- 1 root root  67188 Jan  3 12:49 undionly.kpxe
drwxr-xr-x 2 root root   4096 Jan 19 17:47 x86_64/

root@akshay-VirtualBox:/srv/ltsp# ll /srv/tftp/ltsp/x86_64/
total 39668
drwxr-xr-x 2 root root     4096 Jan 19 17:47 ./
drwxr-xr-x 3 root root     4096 Jan 19 17:48 ../
-rw-r--r-- 1 root root 35324886 Jan 19 17:43 initrd.img
-rw-r--r-- 1 root root  5283056 Nov 28 13:17 vmlinuz


```

**Comment by alkisg** [comment]
At that point in the (initramfs) prompt, what is the output of the following command?

```shell
ls -l /scripts/init-bottom
cat /scripts/init-bottom/ORDER
cat /scripts/init-bottom/ltsp-initrd-bottom
cat /proc/cmdline
```

**Comment by Akshay-Hegde** [comment]
@alkisg Please find attached details

With x86_64.img, its prompting to initramfs prompt
![Screenshot from 2021-01-19 20-30-39](https://user-images.githubusercontent.com/7464492/105053250-be588a80-5a96-11eb-9335-3874331a1748.png)
![Screenshot from 2021-01-19 20-37-16](https://user-images.githubusercontent.com/7464492/105053419-e516c100-5a96-11eb-978d-45a414db6257.png)

However with second option in menu, I am able to login 


![Screenshot from 2021-01-19 20-43-43](https://user-images.githubusercontent.com/7464492/105053902-75550600-5a97-11eb-8ab0-fe4f2c2288a0.png)
![Screenshot from 2021-01-19 20-46-13](https://user-images.githubusercontent.com/7464492/105053985-86057c00-5a97-11eb-9fe9-352fe3d13e58.png)

**Comment by alkisg** [comment]
The problem is that /scripts/init-bottom/ltsp-initrd-bottom is not executable. But the LTSP code does make it executable:

https://github.com/ltsp/ltsp/blob/master/ltsp/client/initrd-bottom/initramfs-tools/ltsp-hook.conf#L84

So it looks like something is very broken in your setup, there's something that cancels our `chmod +x` command. I would need IRC/VNC access to debug this, it can't be done over discussions...

Btw are you sure that you want to use a chroot instead of chrootless or VM?

**Comment by Akshay-Hegde** [comment]
Hi, thanks for the clarification, I am looking for some lightweight image for my thin client with few apps, so tried below, if there's any other easy/better way either by chrootless or VM, please share snippet,  I never built using VM or VMDK or QCOW2.

Even with ubuntu desktop (using chroot) I had no success.

```
mkdir -p /srv/ltsp
cd /srv/ltsp
apt install debootstrap
debootstrap buster x86_64
chroot x86_64
apt update
apt install --no-install-recommends linux-image-amd64 initramfs-tools
apt install xfce4
systemctl set-default graphical.target
passwd
exit
```

**Comment by alkisg** [comment]
A chroot makes sense if you're not going to use a full session, e.g. if you'll just install xorg+freerdp. If you're installing xfce anyway, then this won't really need less RAM than a normal Xubuntu installation. And size doesn't matter as it's not loaded in RAM.

The easiest way would be to install your LTSP server with Xubuntu.iso, and run `ltsp image /`, i.e. the chrootless method. More details in the [ltsp image man page](https://ltsp.org/man/ltsp-image/), which also documents VM images.

If you want help in troubleshooting your existing chroot, I could help you, but you'll need to come over to IRC (go to ltsp.org and click on the IRC link).

**Comment by Akshay-Hegde** [comment]
@alkisg As you suggested with Xubuntu iso, chrootless method worked for me, thanks a ton.

As you said Xubuntu lighter than the method I was trying with debootstrap.

**Comment by alkisg** [comment]
Great! For the next time, please use LTSP discussions: https://github.com/ltsp/discussions
While LTSP issues are only for bugs that need changes in LTSP: https://github.com/ltsp/ltsp

I'll move this issue to discussions later, currently github disabled this functionality due to some bug in their UI.

**Comment by fty1777** [comment]
I met the same problem and finally figured out just installing the `ltsp` package inside the client chroot fix the problem. 

This is because the invocation of ltsp-hook must be written to `initrd.img` by mkinitramfs for taking effect. Installing `ltsp` inside client is a viable and consistent way to trigger mkinitramfs and write the hook into `initrd.img`. Of course one may achieve the same goal through other methods, I just prefer this one for simplicity.

**Reply by fty1777** [reply]
Btw, consider not to put the root inside `/srv/ltsp`, which I suspect would expose sensitive data through TFTP, making the cleanup process in `ltsp image` meaningless.
